### ISSUE TITLE
E3677 'dotnet8' does not match '^(nodejs.*|python.*)$'

### ISSUE THREAD

**Issue by dosolkowski-work** [aws-cloudformation/cfn-lint#4392]
### CloudFormation Lint Version

1.47.0

### What operating system are you using?

Ubuntu

### Describe the bug

cfn-lint believes `dotnet8` is not a valid runtime when using a ZipFile, but it is [supported by AWS Lambda](https://docs.aws.amazon.com/lambda/latest/dg/csharp-package.html) and has been for years.

This started happening in 1.47.0 and did not happen in 1.46.0.

### Expected behavior

Valid Lambda runtimes do not generate an error.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  SomeLambda:
    Type: AWS::Lambda::Function
    Properties:
      Code: path/to/function.zip
      Handler: function.handler
      MemorySize: 1024
      PackageType: Zip
      Role: some:iam:role
      Runtime: dotnet8
      Timeout: 30
      VpcConfig:
        SecurityGroupIds:
          - sg1
        SubnetIds:
          - sn1
          - sn2
```
